### PR TITLE
Unit test/oomstore apply

### PR DESCRIPTION
**Pull request by lianxmfor**
this PR adds unit-test for oomstore/apply

### CODE REVIEW

File: pkg/oomstore/apply.go [hunk 1]
```diff
@@ -196,15 +196,31 @@ func buildApplyStage(ctx context.Context, opt apply.ApplyOpt) (*apply.ApplyStage
 				return nil, err
 			}
 
-			status.NewEntities = append(status.NewEntities, entity)
+			stage.NewEntities = append(stage.NewEntities, apply.Entity{
```
Comment (YikSanChan) [inline pkg/oomstore/apply.go:199]: can you explain why?

**Review by wfxr — APPROVED** [review]
LGTM

File: pkg/oomstore/apply_test.go [hunk 2]
```diff
@@ -0,0 +1,286 @@
...
+						Description: "description",
+					},
+				},
+				NewGroups:   make([]apply.Group, 0),
+				NewFeatures: make([]apply.Feature, 0),
+			},
+			wantErr: nil,
+		},
+		{
+			description: "has many simple object",
```
Comment (YikSanChan) [inline pkg/oomstore/apply_test.go:66]: objects

**Review by YikSanChan — APPROVED** [review]
lgtm

File: pkg/oomstore/apply_test.go (hunk 2 above)
Comment (lianxmfor) [reply pkg/oomstore/apply_test.go]: fixed

**Comment by wfxr** [comment]
https://github.com/oom-ai/oomstore/blob/b7940037444d737cf90a3ade6d45f8f9200e0bcb/pkg/oomstore/apply.go#L192

@lianxmfor I think it's not necessary to make these keys case insensitive ?

File: pkg/oomstore/apply.go (hunk 1 above)
Comment (lianxmfor) [reply pkg/oomstore/apply.go:199]: We don't want `entity.BatchFeature` to have values. The whole stage should be a flat structure, not a nested structure.

Comment (YikSanChan) [reply pkg/oomstore/apply.go:199]: can you add a comment in code? so that people don't ask

**Comment by lianxmfor** [comment]
> I think it's not necessary to make these keys case insensitive ?

hmm. just to avoid user input wrong, improve fault tolerance. if you think this is not good, I can remove it later.

File: pkg/oomstore/apply.go (hunk 1 above)
Comment (lianxmfor) [reply pkg/oomstore/apply.go:201]: sound good, it's done.

**Pull request merged**